helper-cli/src/main/kotlin/commands/GeneratePackageConfigurationsCommand.kt

```diff
@@ -104,7 +106,7 @@ internal class GeneratePackageConfigurationsCommand : CliktCommand(
 }
 
 private fun createPackageConfiguration(id: Identifier, provenance: Provenance): PackageConfiguration =
-    provenance.vcsInfo?.let { vcsInfo ->
+    provenance.asRepository()?.vcsInfo?.let { vcsInfo ->
```

Comment:
I believe these null-safety / Elvis-operator chains are exactly what we aim to get rid of with this refactoring. So how about this instead:

    when (provenance) {
        is RepositoryProvenance -> {
            PackageConfiguration(
                id = id,
                vcs = VcsMatcher(
                    type = provenance.vcsInfo.type,
                    url = provenance.vcsInfo.url,
                    revision = provenance.vcsInfo.resolvedRevision!!,
                    path = provenance.vcsInfo.path.takeIf { provenance.vcsInfo.type == VcsType.GIT_REPO }.orEmpty()
                )
            )
        }

        is ArtifactProvenance -> {
            PackageConfiguration(
                id = id,
                sourceArtifactUrl = provenance.sourceArtifact.url
            )
        }

        else -> throw IllegalArgumentException("Unsupported provenance $provenance.")
    }